backend/Tasque.BLL/Services/UserService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using AutoMapper;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+using Tasque.Core.BLL.Exeptions;
+using Tasque.Core.BLL.JWT;
+using Tasque.Core.Common.DTO;
+using Tasque.Core.Common.Entities;
+using Tasque.Core.Common.Security;
+using Tasque.Core.DAL;
+
+namespace Tasque.Core.BLL.Services
+{
+    public class UserService
+    {
+        private DataContext _context;
+        private IMapper _mapper;
+        private IValidator<User> _validator;
+        private FileUploadService _fileUploadService;
+
+        public UserService(
+            DataContext context,
+            IMapper mapper,
+            IValidator<User> validator,
+            FileUploadService fileUploadService)
+        {
+            _context = context;
+            _mapper = mapper;
+            _validator = validator;
+            _fileUploadService = fileUploadService;
+        }
+
+        public async Task<UserDto> GetUserById(int id)
+        {
+            if (!await _context.Users.AnyAsync(u => u.Id == id))
+            {
+                throw new ValidationException("User with given id does not exist");
+            }
+            var userEntity = await _context.Users.FirstAsync(u => u.Id == id);
+            return _mapper.Map<UserDto>(userEntity);
+        }
+
+        public async Task<UserDto> EditUserProfile(UserDto dto)
+        {
+            var userEntity = await _context.Users.FirstOrDefaultAsync(u => u.Id == dto.Id)
+                ?? throw new ValidationException("User not found");
+            userEntity.Name = dto.Name;
+            if(string.IsNullOrEmpty(dto.AvatarURL) || dto.AvatarURL.StartsWith("http"))
+            {
+                userEntity.AvatarURL = dto.AvatarURL;
+            }
+            else
+            {
+                userEntity.AvatarURL = await _fileUploadService.UploadFileAsync(dto.AvatarURL, "avatars");
```

Comment:
Can you name avatar (as I understand in base64 format)  with more meaningful name? Not AvatarURL

You can create UserEditDto for such purpose

Reply:
I use the same string field for url and file data in base64 format. If it has URL or null, the value srores to db, and if it has file the data saves to S3 and generated link saves to db.